FirebaseStorage/Sources/StorageMetadata.swift

```diff
@@ -29,67 +27,31 @@ import FirebaseStorageInternal
   /**
    * The name of the bucket containing this object.
    */
-  @objc public let bucket: String
+  @objc public let bucket: String?
```

Comment:
API review?

Reply:
Great catch! Fixed and explained in #10239